hack/golangci-lint

```diff
@@ -0,0 +1,22 @@
+#! /usr/bin/env bash
+
+readonly PROGNAME="golangci-lint"
```

Comment:
prefer if the project installs tooling on behalf of the user as shown [here](https://github.com/envoyproxy/gateway/pull/84)
rather than using the `hack` approach